guide/token-setup-and-validation.md

```diff
@@ -0,0 +1,351 @@
+
+//Configure
+//must come before using MVC
+...
+app.UseAuthentication();
+...
+```
+
+### Api controller
+```c#
+// Note: both OpenIddictValidationDefaults.AuthenticationScheme and JwtBearerDefaults.AuthenticationScheme are "Bearer"
+//If "Bearer" is not the default authentication scheme then specify it here.
+//If you get a 302 redirect to login page instead of a 401 Unauthorized then Cookie authentication is handling the request
+//so "Bearer" must be specified
+[Authorize(AuthenticationSchemes = "Bearer")]
+public class MyController : Controller
+```
+
+
+# Reference token format
+
+## Reference token generation
+
+### Authorization server
+
+```c#
+// Startup.cs
+
+//ConfigureServices
+// Register OpenIddict stores
+services.AddDbContext<ApplicationDbContext>(options =>
+    {
+        options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection"));
+        options.UseOpenIddict();
+    });
+
+// Register the OpenIddict services.
+services.AddOpenIddict()
+    .AddCore(options =>
+        {
+            options.UseEntityFrameworkCore()
+                .UseDbContext<ApplicationDbContext>();
+        })
+
+    .AddServer(options => 
+        {
+            //other options left out for brevity
+            options.UseReferenceTokens();
+        })
+
+//Configure
+//must come before using MVC
+...
+app.UseAuthentication();
+...
+```
+
+## Reference token validation
+
+### Authorization server
+```c#
+// Startup.cs
+
+//ConfigureServices
+services.AddOpenIddict()
+        .AddCore(...) //see above
+
+        .AddServer(...) // see above
+
+        .AddValidation(options => options.UseReferenceTokens());
+
+//Configure
+//must come before using MVC
+...
+app.UseAuthentication();
+...
+```
+
+### Resource server
+```c#
+// Startup.cs
+
+//ConfigureServices
+// Register OpenIddict stores
+services.AddDbContext<AuthServerDbContext>(options =>
+    {
+        options.UseSqlServer(Configuration.GetConnectionString("AuthServerConnection"));
+        options.UseOpenIddict();
+    });
+
+
+services.AddOpenIddict()
+    .AddCore(options =>
+    {
+        // Register the Entity Framework entities and stores.
+        options.UseEntityFrameworkCore()
+               .UseDbContext<AuthServerDbContext>();
+    })
+
+    //This adds a "Bearer" authentication scheme
+    .AddValidation(options => options.UseReferenceTokens());
+
+//Optionally set Bearer token authentication as default
+//services.AddAuthentication("Bearer");
+
+//Configure
+//must come before using MVC
+...
+app.UseAuthentication();
+...
+```
+
+### Api controller
+```c#
+// Note: both OpenIddictValidationDefaults.AuthenticationScheme and JwtBearerDefaults.AuthenticationScheme are "Bearer"
+//If "Bearer" is not the default authentication scheme then specify it here.
+//If you get a 302 redirect to login page instead of a 401 Unauthorized then Cookie authentication is handling the request
+//so "Bearer" must be specified
+[Authorize(AuthenticationSchemes = "Bearer")]
```

Comment:
You'll want to use `OpenIddictValidationDefaults.AuthenticationScheme` as the section is dedicated to reference tokens.

Note: in OpenIddict, the default schemes are not meant to be customized.